xla/service/gpu/gpu_executable.cc

```diff
@@ -140,7 +140,7 @@ GpuExecutable::GpuExecutable(GpuExecutable::Params params)
 #if TENSORFLOW_USE_ROCM
   // ROCm uses hsaco hashes to distinguish between modules.
   // Bad things happen if multiple modules with identical code are loaded.
-  binary_.reserve(binary_.size() + 272);
+  binary_.reserve(binary_.size() + 16);
```

Comment:
So to make it equivalent to your original change, I would suggest to use:

binary_.reserve(binary_.size() + 272);
binary_.resize(binary_.size() + 16);

But if you don't need the extra 256 bytes, then just use
binary_.resize(binary_.size() + 16);

Reply:
Thanks a lot! Yes, we don't need ROCm3.3 part anymore. 

We don't need resize() as it also creates instances, reserver(binary_.size() + 16) that only allocates memory and it has passed all our internal CI passes.

Reply:
The question is not whether it passes all tests, but whether it is good C++ code. Now if you just reserve the memory, but do not increase the size of the vector, think about what happens if you later would resize the vector: it would just overwrite those values you wrote "out of bounds". See also some examples here:

http://www.gotw.ca/gotw/074.htm

If you worry about the extra initialization costs of resize, you can use push_back instead of writing to the reserved memory locations out of bounds.
So summarizing: please either use resize instead of reserve, or reserve + push_back for adding the extra values.

Reply:
I c your point. Thanks a lot @akuegel for patiently explainning and reference! I have changed to resize().